### PR TITLE
start of assignment 4

### PR DESCRIPTION
**Pull request by taylorpitman**
I got to Functional Requirements 2.. Having trouble connecting the components. and getting confused on what I have to implement vs test implement.

### CODE REVIEW

File: .github/src/EngineManager.java [hunk 1]
```diff
@@ -1,15 +1,73 @@
...
+ * 
+ *      d. Request that the data storage component write the results to the output
  */
 
+import java.util.List;
+
 public class EngineManager implements ComputeAPI {
+
     DataAPI dataStoreAPI;
     EngineAPI computeEngineAPI;
```
Comment (jcalhounn) [inline .github/src/EngineManager.java]: Having the EngineManager constructor seems off. The User has an instance variable that has access to the methods inside of the "ComputeAPI" interface. 

The "ComputeAPI" interface requires that the EngineManager can handle the "compute" method that takes a ComputeRequest variable type.

I believe that the arguments in the "EngineManager" constructor should be empty.

File: .github/src/EngineManager.java [hunk 2]
```diff
@@ -1,15 +1,73 @@
...
+    */
     @Override
     public ComputeResult compute(ComputeRequest request) {
-        // TODO Auto-generated method stub
-        throw new UnsupportedOperationException("Unimplemented method 'compute'");
+        
+        //request DataAPI object to read ints
+        request = new ComputeRequestTestImpl();
+        InputConfig inputConfig = request.getInputConfig();
+        DataReadResult readRequest = dataStoreAPI.read(inputConfig);
```
Comment (jcalhounn) [inline .github/src/EngineManager.java]: I think that line 42 and 43 could be condensed down to : DataReadResult readRequest = dataStoreAPI.read(request.getInputConfig());

this method call would have the correct return type to pass into the arguments of the "read" method call.

File: .github/src/EngineManager.java [hunk 3]
```diff
@@ -1,15 +1,73 @@
...
+
+        if(readRequest.equals(DataReadResult.SUCCESS))
+        {
+            
+            
+            //EngineComputeRequest computeRequest = new EngineComputeRequestImpl(list);
+
+            //pass the ints to the EngineAPI object
+            EngineComputeResult computeResult = computeEngineAPI.compute(list); //THIS is where I am stuck on what to do next
+
```
Comment (jcalhounn) [inline .github/src/EngineManager.java:54]: I do not think that the EngineManager should have an "EngineComputeResult" type variable.

Similarly to how the User has an instance variable of ComputeAPI that EngineManager implements, I believe that the EngineManager should have an instance variable from the EngineAPI that the EngineCompute will handle.

File: .github/src/DataReadResult.java [hunk 4]
```diff
@@ -1,6 +1,7 @@
 //decides how to model the arbitrarily-large number of integer inputs
```
Comment (jcalhounn) [inline .github/src/DataReadResult.java:1]: I believe this interface should solely be responsible for determining SUCCESS or FAILURE for the data read attempt by "DataStorage"

**Review by taylorpitman — COMMENTED** [review]
peer reviewed in person

**Review by jcalhounn — APPROVED** [review]
discussed in person

**Pull request merged**